src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -229,10 +230,27 @@ private String href(String path, boolean isDir) {
         return href;
     }
 
-    public String formatHtmlLink(String link) {
+    private static String formatHtmlLink(String link) {
         return link.replaceAll("&", "&amp;");
     }
 
+    private static String formatPathURL(String input) {
```

Comment:
I'd suggest we keep our own method.. It is specific enough to our case to justify it imo. I have improved it though to make sure every character is escaped.